src/logging.cpp

```diff
@@ -389,23 +390,10 @@ void BCLog::Logger::LogPrintStr(std::string_view str, std::string_view logging_f
 
 void BCLog::Logger::LogPrintStr_(std::string_view str, std::string_view logging_function, std::string_view source_file, int source_line, BCLog::LogFlags category, BCLog::Level level)
 {
+    Assume(str.ends_with("\n")); // Checked at compile-time, so this can never fail
```

Comment:
> `Assume` should be used to document assumptions when program execution can safely continue even if the assumption is violated.

Shouldn't this be an `assert` instead?

Reply:
It seems a little crazy for logging code to use `\n` in format strings when logger doesn't support multiline log strings, and more crazy to add a compile time check enforcing presence of `\n` characters which serve no purpose, add noise, and give misleading impression that the logger supports multiline strings when it doesn't. I think the best thing would be go to ignore trailing \n characters, stop using them going forward, and not require them.